### ISSUE TITLE
The dict data type cann't be accessed by dot '.'

### ISSUE THREAD

**Issue by lauraset** [Junjue-Wang/LoveDA#44]
Hello, @Junjue-Wang. I want to know whether you have rewrited the dict data type. I cann't accessed the members of dict by dot '.'.
For example, for the SOURCE_DATA_CONFIG, when I try to obtain the value of 'image_dir',  `SOURCE_DATA_CONFIG['image_dir']` can work, but `SOURCE_DATA_CONFIG.image_dir` can not.
![image](https://user-images.githubusercontent.com/39206462/186302818-ee957c3e-2ecb-4aaa-9b32-e6db872d1803.png)
When you define `LoveDALoader`, the dot, e.g., `config.batch_size` can obtain the value of dict members. How to do it? Thank you in advance.

![image](https://user-images.githubusercontent.com/39206462/186303220-e19b3e06-1bee-4108-963b-9b4333be0b28.png)

**Comment by Junjue-Wang** [comment]
Yes, we have re-implemented the dict for config

**Comment by lauraset** [comment]
@Junjue-Wang, Thank you very much. I still have one question about the `clip_grad.clip_grad_norm_(filter(lambda p: p.requires_grad, model.parameters()), max_norm=35, norm_type=2)`. I guess the `max_norm` should be changed according to different datasets. Whether it is necessary for other new datasets? Thank you for your reply in advance.

**Comment by Junjue-Wang** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        I think this is not sensitive to specific datasets
        
    
        
            
        
    
    
     
    
        
                    ***@***.***
        
    
 

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        8/25/2022 16:31
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [Junjue-Wang/LoveDA] The dict data type cann't be accessed by dot '.' (Issue #44)
        
      
    
  
  
@Junjue-Wang, Thank you very much. I still have one question about the clip_grad.clip_grad_norm_(filter(lambda p: p.requires_grad, model.parameters()), max_norm=35, norm_type=2). I guess the max_norm should be changed according to different datasets. Whether it is necessary for other new datasets? Thank you for your reply in advance.

**Comment by lauraset** [comment]
Oh, thank you anyway. I just want to know why `max_norm` is set to `35` . Do you set it by experiments?

**Comment by Junjue-Wang** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        
            
        
    
    
        No, it is the default setting
        
    
     
    
        
                    ***@***.***
        
    
 

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        8/25/2022 17:13
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [Junjue-Wang/LoveDA] The dict data type cann't be accessed by dot '.' (Issue #44)
        
      
    
  
  
Oh, thank you anyway. I just want to know why max_norm is set to 35 . Do you set it by experiments?

**Comment by lauraset** [comment]
Oh, I see. Thank you a lot.
